python/pyspark/sql/tests/test_utils.py

```diff
@@ -287,25 +302,30 @@ def test_assert_notequal_arraytype(self):
             ),
         )
 
-        expected_error_message = "Results do not match: "
-        percent_diff = (1 / 2) * 100
-        expected_error_message += "( %.5f %% )" % percent_diff
+        if isinstance(df2, DataFrame):
+            actual_str = df1._jdf.showString(2, 2, False)
+            expected_str = df2._jdf.showString(2, 2, False)
+        else:
+            # Spark Connect
+            actual_str = df1._show_string(2, 2, False)
+            expected_str = df2._show_string(2, 2, False)
```

Comment:
Instead of using _show_string to invoke a Spark job, how about we convert the df to a pandas dataframe. Pandas should be a required dependency when spark connect is enabled.